kafka/src/main/java/com/expedia/adaptivealerting/kafka/KafkaMultiClusterAnomalyToMetricMapper.java

```diff
@@ -17,124 +17,109 @@
-    private final MetricTankIdFactory metricTankIdFactory = new MetricTankIdFactory();
+    private String anomalyTopic;
+    private String metricTopic;
     
     public static void main(String[] args) {
+        
         // TODO Refactor the loader such that it's not tied to Kafka Streams. [WLW]
         val config = new TypesafeConfigLoader(APP_ID).loadMergedConfig();
-        val mapper = new KafkaMultiClusterAnomalyToMetricMapper(config);
+        
+        val consumerConfig = config.getConfig(ANOMALY_CONSUMER);
+        val consumerTopic = consumerConfig.getString(TOPIC);
+        val consumerProps = ConfigUtil.toConsumerConfig(consumerConfig);
+        val consumer = new KafkaConsumer<String, MappedMetricData>(consumerProps);
+        
+        val producerConfig = config.getConfig(METRIC_PRODUCER);
+        val producerTopic = producerConfig.getString(TOPIC);
+        val producerProps = ConfigUtil.toProducerConfig(producerConfig);
+        val producer = new KafkaProducer<String, MetricData>(producerProps);
+        
+        val mapper = new KafkaMultiClusterAnomalyToMetricMapper(consumer, producer, consumerTopic, producerTopic);
         mapper.run();
     }
     
-    public KafkaMultiClusterAnomalyToMetricMapper(Config config) {
-        this.anomalyConsumer = createAnomalyConsumer(config.getConfig("anomaly-consumer"));
-        this.metricProducer = createMetricProducer(config.getConfig("metric-producer"));
+    public KafkaMultiClusterAnomalyToMetricMapper(
+            Consumer<String, MappedMetricData> anomalyConsumer,
+            Producer<String, MetricData> metricProducer,
+            String anomalyTopic,
+            String metricTopic) {
+        
+        this.anomalyConsumer = anomalyConsumer;
+        this.metricProducer = metricProducer;
+        this.anomalyTopic = anomalyTopic;
+        this.metricTopic = metricTopic;
     }
     
     @Override
     public void run() {
         log.info("Starting KafkaMultiClusterAnomalyToMetricMapper");
-        anomalyConsumer.subscribe(Collections.singletonList(anomaliesTopic));
+        anomalyConsumer.subscribe(Collections.singletonList(anomalyTopic));
         
-        // TODO Implement clean shutdown.
         // See Kafka: The Definitive Guide, pp. 86 ff.
-        while (true) {
-            try {
-                anomalyConsumer.poll(100).forEach(record -> {
-                    if (record != null) {
-                        metricProducer.send(toMetricDataRecord(record));
-                    }
-                });
-            } catch (Exception e) {
-                log.error("Exception while processing anomalies", e);
+        try {
+            while (true) {
+                val records = anomalyConsumer.poll(POLL_PERIOD);
+                log.trace("numRecords={}", records.count());
+                records.forEach(record -> metricProducer.send(toMetricDataRecord(record)));
```

Comment:
Are we no-longer worried about breaking from the while loop? Do we just wait for app to restart? What if we have a bad record on the topic